test/rihanna/job_dispatcher_test.exs

```diff
@@ -216,12 +216,19 @@ defmodule Rihanna.JobDispatcherTest do
       assert is_pid(state.pg)
     end
 
+    test "runs the after_error callback on the job when it returns {:error, reason}" do
+      ref = make_ref()
+      {:ok, _} = Rihanna.Job.enqueue({ErrorTupleBehaviourMock, [self(), ref]})
+      :timer.sleep(100)
+      assert_received "After error callback"
+    end
+
     test "marks job as failed when returns :error", %{pg: pg} do
       ref = make_ref()
       {:ok, %{id: id}} = Rihanna.Job.enqueue({ErrorBehaviourMock, [self(), ref]})
       :timer.sleep(100)
       %Rihanna.Job{fail_reason: reason} = get_job_by_id(pg, id)
-      assert reason == "Job Failed\n{:error, \"failed for some reason\"}"
+      assert reason == ":error"
```

Comment:
The message here is gone because the mock was returning the tuple when it should have only been the atom

Reply:
This should probably be a separate PR since it changes the external interface.